File: Sprint8/G/C++/main.cpp
// https://contest.yandex.ru/contest/26131/problems/G/
#include <iostream>
#include <vector>

bool check_equal(const std::vector<int>& s, int start, const std::vector<int>& pattern, int shift) {
    for (int i = 0; i < pattern.size(); ++i) {
        if (s[start + i] != pattern[i] + shift) {
            return false;
        }
    }
    return true;
}

int main() {
    int n, m;
    std::cin >> n;
    std::vector<int> measure_res(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> measure_res[i];
    }

    std::cin >> m;
    std::vector<int> sample(m);
    for (int i = 0; i < m; ++i) {
        std::cin >> sample[i];
    }

    for (int i = 0; i <= n - m; ++i) {
        int shift = measure_res[i] - sample[0];
        if (check_equal(measure_res, i, sample, shift)) {
            std::cout << i + 1 << " ";
        }
    }

    return 0;
}

File: Sprint8/G/Rust/src/main.rs
// https://contest.yandex.ru/contest/26131/problems/G/
use std::io::BufRead;

fn check_equal(s: &[i32], start: usize, pattern: &Vec<i32>, shift: i32) -> bool {
    for i in 0..pattern.len() {
        if s[start + i] != pattern[i] + shift {
            return false;
        }
    }
    true
}

fn main() {
    let mut lines = std::io::stdin().lock().lines();
    let n: usize = lines.next().unwrap().unwrap().parse().unwrap();
    let mut measure_res = Vec::with_capacity(n);

    measure_res.extend(
        lines
            .next()
            .unwrap()
            .unwrap()
            .split_whitespace()
            .map(|s| s.parse::<i32>().unwrap()),
    );

    let m: usize = lines.next().unwrap().unwrap().parse().unwrap();
    let mut sample = Vec::with_capacity(m);

    sample.extend(
        lines
            .next()
            .unwrap()
            .unwrap()
            .split_whitespace()
            .map(|s| s.parse::<i32>().unwrap()),
    );

    for i in 0..=n - m {
        let shift = measure_res[i] - sample[0];
        if check_equal(&measure_res, i, &sample, shift) {
            print!("{} ", i + 1);
        }
    }
}
